feat: add sso (enterprise login) docs under platform (#10924)
supabase/supabase: apps/docs/pages/guides/platform/sso/sso-with-gsuite.mdx

import Layout from '~/layouts/DefaultGuideLayout'

export const meta = {
  id: 'sso-with-gsuite',

>   title: 'Setup SSO with Google Workspace',
>   description: 'Configuration instructions for single sign-on with Google Workspace (GSuite)',

Reviewer: ```suggestion
  title: 'Set Up SSO with Google Workspace',
  description: 'Configure single sign-on with Google Workspace (GSuite).',
```

}

Supabase supports single sign-on (SSO) using Google Workspace (formerly known
as GSuite).

> ## Step-by-step guide
> 
> ### Step 1: Open the Google Workspace Web and mobile apps console

Reviewer: ```suggestion
## Step 1: Open the Google Workspace Web and mobile apps console [#google-workspace-console]
```

![Google Workspace: Web and mobile apps admin
console](/docs/img/sso-gsuite-step-01.png)

> ### Step 2: Choose Add custom SAML app

Reviewer: ```suggestion
## Step 2: Choose Add custom SAML app [#add-custom-saml-app]
```

From the *Add app* button in the toolbar choose *Add custom SAML app*.

![Google Workspace: Web and mobile apps admin console, Add custom SAML app
selected](/docs/img/sso-gsuite-step-02.png)

> ### Step 3: Fill out App details

Reviewer: ```suggestion
## Step 3: Fill out app details [#add-app-details]
```

The information you enter here is for visibility into your Google Workspace.
You can choose any values you like. `Supabase` as a name works well for most
use cases. Optionally enter a description.

![Google Workspace: Web and mobile apps admin console, Add custom SAML, App
details screen](/docs/img/sso-gsuite-step-03.png)

> ### Step 4: Download IdP metadata

Reviewer: ```suggestion
## Step 4: Download IdP metadata [#download-idp-metadata]
```

This is a very important step. Click on *DOWNLOAD METADATA* and save the file
that was downloaded.

![Google Workspace: Web and mobile apps admin console, Add custom SAML, Google
Identity Provider details screen](/docs/img/sso-gsuite-step-04.png)

It's very important to Send this file to your support contact at Supabase to
complete the SSO setup process. If you're not sure where to send this file, you
can always reach us at [support@supabase.com](mailto:support@supabase.com).

**Important: Make sure the certificate as shown on screen has at least 1 year
before it expires. Mark down this date in your calendar so you will be reminded
that you need to update the certificate without any downtime for your users.**

> ### Step 5: Add service provider details

Reviewer: ```suggestion
## Step 5: Add service provider details [#add-service-provider-details]
```

Fill out these serivce provider details on the next screen.

| Detail | Value |
| ------ | ----- |
| ACS URL | `https://alt.supabase.io/auth/v1/sso/saml/acs` |
| Entity ID | `https://alt.supabase.io/auth/v1/sso/saml/metadata` |
| Start URL | `https://app.supabase.com` |
| Name ID format | PERSISTENT |
| Name ID | *Basic Information > Primary email* |

![Google Workspace: Web and mobile apps admin console, Add custom SAML,
Service provider details screen](/docs/img/sso-gsuite-step-05.png)

> ### Step 6: Configure Attribute mapping

Reviewer: ```suggestion
## Step 6: Configure Attribute mapping [#configure-attribute-mapping]
```

Attribute mappings allow Supabase to get information about your Google
Workspace users on each login.

**A *Primary email* to `email` mapping is required to exist.** Other mappings
shown below are optional and configurable depending on your Google Workspace
setup. If in doubt, replicate the same config as shown.

Please share any changes, if any, from this screen with your Supabase support
contact.

![Google Workspace: Web and mobile apps admin console, Add custom SAML,
Attribute mapping](/docs/img/sso-gsuite-step-06.png)

> ### Step 7: Wait for confirmation

Reviewer: ```suggestion
## Step 7: Wait for confirmation [#confirmation]
```

Once you’ve configured the Google Workspace app as shown above, make sure you

> send the metadata file you downloaded from [Step
> 4](#step-4-download-idp-metadata) and information regarding the
> attribute mapping from [Step 6](#step-6-configure-attribute-mappings) (if any

Reviewer: ```suggestion
send the [metadata file you downloaded](#download-idp-metadata)
and information regarding the [attribute mapping](#configure-attribute-mappings) (if any
```

changes are applicable) to your support contact at Supabase.

This information needs to be entered into Supabase before SSO is activated
end-to-end.

Wait for confirmation that this information has successfully been added to
Supabase. It usually takes us 1 business day to configure this information
for you.
Supabase.

> ### Step 8: Configure user access

Reviewer: ```suggestion
## Step 8: Configure user access  [#configure-user-access]
```

You can configure which Google Workspace user accounts will get access to
Supabase. This is important if you wish to limit access to your software
engineering teams.

You can configure this access by clicking on the *User
access* card (or down-arrow). Follow the instructions on screen.

Changes from this step sometimes take a while to propagate across Google’s
systems. Please wait at least 15 minutes before proceeding to the next step.

![Google Workspace: Web and mobile apps admin console, Supabase app
screen](/docs/img/sso-gsuite-step-08.png)

> ### Step 9: Test single sign-on

Reviewer: ```suggestion
## Step 9: Test single sign-on [#testing]
```
